frontend/src/pages/NewRunV2.tsx

```diff
@@ -81,89 +86,25 @@ function NewRunV2(props: NewRunV2Props) {
   const [errorMessage, setErrorMessage] = useState('');
   const [isParameterValid, setIsParameterValid] = useState(false);
   const [clonedRuntimeConfig, setClonedRuntimeConfig] = useState<PipelineSpecRuntimeConfig>({});
-  const [usePipelineFromRunLabel, setUsePipelineFromRunLabel] = useState(
-    'Using pipeline from cloned run',
-  );
 
   // TODO(zijianjoy): If creating run from Experiment Page or RunList Page, there is no pipelineId/Version.
   const urlParser = new URLParser(props);
-  const originalRunId = urlParser.get(QUERY_PARAMS.cloneFromRun);
-  const pipelineId = urlParser.get(QUERY_PARAMS.pipelineId);
-  const pipelineVersionIdParam = urlParser.get(QUERY_PARAMS.pipelineVersionId);
-  const pipelineDetailsUrl = originalRunId
+  const usePipelineFromRunLabel = 'Using pipeline from existing run.';
+  const { apiRun, apiPipeline, apiPipelineVersion } = props;
+  const pipelineDetailsUrl = props.originalRunId
     ? RoutePage.PIPELINE_DETAILS.replace(
         ':' + RouteParams.pipelineId + '/version/:' + RouteParams.pipelineVersionId + '?',
         '',
-      ) + urlParser.build({ [QUERY_PARAMS.fromRunId]: originalRunId })
+      ) + urlParser.build({ [QUERY_PARAMS.fromRunId]: props.originalRunId })
     : '';
 
-  // Retrieve Run Detail using RunID from backend
-  const { data: apiRun } = useQuery<ApiRunDetail, Error>(
-    ['ApiRun', originalRunId],
-    () => {
-      if (!originalRunId) {
-        throw new Error('No Run ID');
-      }
-      const originalRun = Apis.runServiceApi.getRun(originalRunId);
-      return originalRun;
-    },
-    { enabled: !!originalRunId, staleTime: Infinity },
-  );
+  const templateString = props.templateString;
+  const isTemplatePullSuccess = templateString !== '';
+  const apiResourceRefFromRun = apiRun ? apiRun.run?.resource_references : undefined;
```

Comment:
NIT: Should we use the following condition?

```
const apiResourceRefFromRun = apiRun?.run?.resource_references ? apiRun.run.resource_references : undefined;
```

Reply:
Yes, because apiRun might be `undefined` (when creating a new run from pipeline ID).

Reply:
My suggestion is different from your current code. If you agree with my proposal, please apply this change.